app/models/authenticator/base.rb

```diff
@@ -137,7 +137,18 @@ def authorize(taskid, username, *args)
           end
 
           user.lastlogon = Time.now.utc
-          user.save!
+          if user.new_record?
```

Comment:
I like that we only lock on a new user, which should ideally be less common than just logging in.  This is great!